packages/api-graphql/src/internals/APIClient.ts

```diff
@@ -721,6 +721,44 @@ export function generateGraphQLDocument(
 		selectionSet as ListArgs['selectionSet'],
 	);
 
+	// default PK args for get and list operations
+	// modified below for CPK
+	const getPkArgs = {
+		[primaryKeyFieldName]: `${fields[primaryKeyFieldName].type}!`,
+	};
+	const listPkArgs = {};
+
+	const generateSkArgs = (op: 'get' | 'list') => {
+		return sortKeyFieldNames.reduce(
+			(acc: Record<string, any>, fieldName: string) => {
+				const fieldType = fields[fieldName].type;
+
+				if (op === 'get') {
+					acc[fieldName] = `${fieldType}!`;
+				} else if (op === 'list') {
+					acc[fieldName] = `Model${fieldType}KeyConditionInput`;
+				}
+
+				return acc;
+			},
+			{},
+		);
+	};
```

Comment:
nit: can use `Object.fromEntries` to save a few bytes :D (still over the limit a bit though...)
```suggestion
	const generateSkArgs = (op: 'get' | 'list') =>
		Object.fromEntries(
			sortKeyFieldNames.map(fieldName => {
				const fieldType = fields[fieldName].type;

				return [
					fieldName,
					op === 'get' ? `${fieldType}!` : `Model${fieldType}KeyConditionInput`,
				];
			}),
		);
```

Reply:
I think this is hard to read/grok. I'd rather keep it very simple since we've had lots of bugs in this code already.